godot-core/src/builtin/collections/packed_array.rs

```diff
@@ -433,10 +452,38 @@ macro_rules! impl_packed_array {
             }
         }
 
+        #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given Rust array.")]
+        impl<const N: usize> From<[$Element; N]> for $PackedArray {
+            fn from(arr: [$Element; N]) -> Self {
+                let mut ret = match N {
+                    0 => return Self::new(),
+                    len => Self::default_with_size(len),
+                };
+
+                // SAFETY: The packed array contains exactly N elements and the source array will be forgotten.
+                unsafe {
+                    let arr = std::mem::ManuallyDrop::new(arr);
+                    ret.move_from_slice(arr.as_ptr(), N);
+                }
```

Comment:
I think it would be more explicit to call `std::mem::forget()` on the array after moving. This should be outside the unsafe block.

Reply:
If `move_from_slice` panics then there's risk of double-free. `ManuallyDrop` is a bit more automatic.

Reply:
We should probably try to make `move_from_slice` not panic under almost any circumstance (except for potential `debug_asserts`), but good point! Can you write this in a comment? To make sure it's not refactored away in the future.